### PR TITLE
Bug 1728808 - Implement a PlatformInfo module that works in Node.js

### PR DESCRIPTION
**Pull request by brizental**
This is what the ping looks like when sent from Node.js in my computer.

```json
(Glean.core.Pings.Maker) {
  "ping_info": {
    "seq": 0,
    "start_time": "2021-09-03T12:08+02:00",
    "end_time": "2021-09-03T12:08+02:00"
  },
  "client_info": {
    "telemetry_sdk_build": "0.19.0",
    "client_id": "7f7aacd9-7d4d-4ca5-bc69-2ab9f2705e08",
    "first_run_date": "2021-09-03+02:00",
    "os": "Darwin",
    "os_version": "20.6.0",
    "architecture": "x64",
    "locale": "en-US",
    "app_build": "Unknown",
    "app_display_version": "Unknown"
  }
}
```

### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [ ] ~**Tests**: This PR includes thorough tests or an explanation of why it does not~
   - Not really sure how to write a test for this, but I guess in this manual testing is enough.
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [ ] ~**Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work~

### CODE REVIEW

File: glean/src/core/platform_info.ts [hunk 1]
```diff
@@ -13,14 +13,19 @@ export const enum KnownOperatingSystems {
   NetBSD = "NetBSD",
   OpenBSD = "OpenBSD",
   Solaris = "Solaris",
+  Unknown = "Unknown",
```
Comment (Dexterp37) [inline glean/src/core/platform_info.ts:16]: Why was this moved here? Wasn't it intentional to leave it as the last?

File: glean/src/platform/node/index.ts [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// Placeholding with the Test Platform for now
-// until we implement each component of the Node platform.
-export { default } from "../test/index.js";
+import PlatformInfo from "./platform_info.js";
+import type Platform from "../index.js";
+
+// We will still use the TestPlatform as a placeholder
+// for the other Node.js modules that have not been implemented yet.
```
Comment (Dexterp37) [inline glean/src/platform/node/index.ts:9]: If you filed bugs for these, would you like to mention them here?

File: glean/src/platform/node/platform_info.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      return Promise.resolve(KnownOperatingSystems.AIX);
+    default:
+      return Promise.resolve(KnownOperatingSystems.Unknown);
+    }
+  },
+
+  async osVersion(): Promise<string> {
+    // Note: `os.release()` returns the _kernel_, not OS version.
+    // That is the best Node.js will give us, so we will take it.
+    return Promise.resolve(os.release() || "Unknown");
```
Comment (Dexterp37) [inline glean/src/platform/node/platform_info.ts:36]: What did @badboy end up picking for the values? Unknown or unknown?

Comment (brizental) [reply glean/src/platform/node/platform_info.ts:36]: "Unkown"

File: glean/src/core/platform_info.ts (hunk 1 above)
Comment (brizental) [reply glean/src/core/platform_info.ts:16]: I moved it so it is toghether with the other values that glean-core also reports. If it was intentional that it was the last I don't remember, heh.
